scipy/interpolate/interpnd.pyx

```diff
@@ -929,17 +982,18 @@ class CloughTocher2DInterpolator(NDInterpolatorBase):
         cdef const double_or_complex[:,:,:] grad = self.grad
         cdef double_or_complex[:,::1] out
         cdef const int[:,::1] simplices = self.tri.simplices
-        cdef double c[NPY_MAXDIMS]
+        cdef double[:,:] c
         cdef double_or_complex f[NPY_MAXDIMS+1]
         cdef double_or_complex df[2*NPY_MAXDIMS+2]
         cdef double_or_complex w
         cdef double_or_complex fill_value
-        cdef int i, j, k, ndim, isimplex, start, nvalues
+        cdef int i, j, k, ndim, isimplex, nvalues
         cdef qhull.DelaunayInfo_t info
-        cdef double eps, eps_broad
+        cdef int[:] isimplices
+
+        isimplices,c = self._find_simplicies(xi)
```

Comment:
This was in a nogil section, any specific reason for moving it out

Reply:
Every part of this function which was inside a `nogil` clause is still inside one, there is a `nogil` around the main loop of the function.